main.go

```diff
@@ -26,8 +26,13 @@ var yarnHistoryDir = flag.String("yarn-history-dir", "/tmp/staging/history/done"
 var httpTimeout = flag.Duration("http-timeout", time.Second*2, "The timeout used for connecting to YARN API. Pass values like: 2s")
 var pollInterval = flag.Duration("poll-interval", time.Second*5, "How often should we poll the job APIs. Pass values like: 2s")
 var enableDebug = flag.Bool("pprof", false, "Enable pprof debugging tools at /debug.")
+var s3BucketName = flag.String("s3-bucket", "", "S3 bucket to fetch old jobs from")
+var s3Region = flag.String("s3-region", "", "AWS region for the job storage S3 bucket")
+var s3JobsPrefix = flag.String("s3-jobs-prefix", "", "S3 key prefix (\"folder\") where jobs are stored")
```

Comment:
what do you think of asserting this is a non-empty string if the s3-region/s3-bucket are non-empty strings? a potential oversight here could potentially flood the bucket with job data at the top-level (although maybe that's desired in some non-Stripe configurations).

Reply:
hmm yeah i think no matter what we do you'll be able to shoot yourself in the foot with this. also it's set to retrieve 1000 keys max by default, so it will be more like a considerable slowdown than a meltdown